2018-04-21 11:22:59,508 ERROR: Exception on /index [GET] [in /Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/app.py:1560]
Traceback (most recent call last):
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/app/routes.py", line 38, in index
    next_url=next_url, prev_url=prev_url)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/templating.py", line 134, in render_template
    context, ctx.app)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/templating.py", line 116, in _render
    rv = template.render(context)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/app/templates/index.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/app/templates/base.html", line 31, in top-level template code
    {% block content %}{% endblock %}
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/app/templates/index.html", line 7, in block "content"
    {{  form.hidden.tag() }}
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/jinja2/environment.py", line 430, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'app.forms.PostForm object' has no attribute 'hidden'
2018-04-21 11:29:12,939 INFO: Microblog startup [in /Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/app/__init__.py:33]
2018-04-21 11:29:56,965 ERROR: Exception on /explore [GET] [in /Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/app.py:1560]
Traceback (most recent call last):
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/venv/lib/python3.6/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/app/routes.py", line 145, in explore
    posts = Post.query.order_by(Post.timestamp.desc()).paginate(page, app.config['POSTS_PER_PAGE', False])
KeyError: ('POSTS_PER_PAGE', False)
2018-04-21 11:35:04,081 INFO: Microblog startup [in /Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/app/__init__.py:33]
2018-04-30 18:58:27,145 INFO: Microblog startup [in /Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/app/__init__.py:33]
2018-05-01 18:28:55,588 INFO: Microblog startup [in /Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/app/__init__.py:34]
2018-05-01 18:30:10,684 INFO: Microblog startup [in /Users/imranh/Documents/Programming/PythonTraining/FlaskTutorial/microblog/app/__init__.py:34]
